src/main/kotlin/no/fdk/fdk_reasoning_service/config/CircuitBreakerConsumerConfiguration.kt

```diff
@@ -0,0 +1,53 @@
+package no.fdk.fdk_reasoning_service.config
+
+import io.github.resilience4j.circuitbreaker.CircuitBreaker.StateTransition
+import io.github.resilience4j.circuitbreaker.CircuitBreakerRegistry
+import io.github.resilience4j.circuitbreaker.event.CircuitBreakerOnStateTransitionEvent
+import no.fdk.fdk_reasoning_service.kafka.KafkaManager
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+open class CircuitBreakerConsumerConfiguration(
+    private val circuitBreakerRegistry: CircuitBreakerRegistry,
+    private val kafkaManager: KafkaManager
+) {
+    init {
+        LOGGER.debug("Configuring circuit breaker event listener")
+        val topics = listOf(
+            "dataset-reasoning",
+            "concept-reasoning",
+            "data-service-reasoning",
+            "information-model-reasoning",
+            "service-reasoning",
+            "event-reasoning"
+        )
+        topics.forEach { topic ->
+            circuitBreakerRegistry.circuitBreaker(topic)
+                .eventPublisher
+                .onStateTransition { event: CircuitBreakerOnStateTransitionEvent ->
+                    handleStateTransition(event, topic)
+                }
+        }
```

Comment:
da kan du feks gjøre det sånn som dette:
```suggestion
        val listener = "reasoning"
        circuitBreakerRegistry.circuitBreaker(listener)
            .eventPublisher
            .onStateTransition { event: CircuitBreakerOnStateTransitionEvent ->
                handleStateTransition(event, listener)
            }
```